Packages/Tracking/Core/Runtime/Scripts/XRHandsSubsystem/Scripts/LeapHandsSubsystem.cs

```diff
@@ -0,0 +1,170 @@
+using Unity.Collections;
+using UnityEngine;
+using UnityEngine.XR.Hands;
+using UnityEngine.SubsystemsImplementation;
+using UnityEngine.XR.Hands.ProviderImplementation;
+
+using Leap;
+using Leap.Unity;
+using System;
+using static UnityEngine.XR.Hands.XRHandSubsystem;
+using UnityEngine.XR.Hands.OpenXR;
+
+namespace Leap.Unity
+{
+    class LeapXRHandProvider : XRHandSubsystemProvider
+    {
+        LeapProvider provider;
+
+        public override void Destroy()
+        {
+        }
+
+        public override void GetHandLayout(NativeArray<bool> handJointsInLayout)
+        {
+        }
+
+        public override void Start()
+        {
+            provider = GameObject.FindObjectOfType<XRLeapProviderManager>();
+        }
+
+        public override void Stop()
+        {
+        }
+
+        static internal string id { get; private set; }
+
+        static LeapXRHandProvider() => id = "UL XR Hands";
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        static void Register()
+        {
+
+            ///This needs to be implemented here when we have the subsystem in place
+            //var handsSubsystemCinfo = new XRHandSubsystemDescriptor.Cinfo
+            //{
+            //    id = id,
+            //    providerType = typeof(OpenXRHandProvider)
+            //};
+            //XRHandSubsystemDescriptor.Register(handsSubsystemCinfo);
+            
+        }
```

Comment:
Is this one necessary? - should we have the descriptor registration here, or in the LeapHandSubsystem?